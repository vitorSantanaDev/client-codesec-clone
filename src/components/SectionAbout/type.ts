import { ISectionAboutProps } from 'interfaces/sections'

export type IViewProps = ISectionAboutProps
