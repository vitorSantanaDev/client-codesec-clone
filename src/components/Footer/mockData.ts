export const mockDataFooterProps = {
  logo: {
    altText: 'logo codesec',
    urlImage: '/images/logo-2-codesec.svg'
  },
  footerContent: 'Todos od direitos reservados',
  links: [
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/instagram.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/facebook.svg'
      }
    },
    {
      href: '/',
      icon: {
        altText: 'icon social',
        urlImage: '/images/youtube.svg'
      }
    }
  ]
}
