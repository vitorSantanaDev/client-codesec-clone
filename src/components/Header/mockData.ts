import { IViewProps } from './type'

export const mockHeaderProps: IViewProps = {
  logo: {
    urlImage: '/images/codesec-logo.svg',
    altText: 'Logo codesec'
  },
  button: {
    url: '/',
    buttonText: 'Inscreva-se'
  }
}
