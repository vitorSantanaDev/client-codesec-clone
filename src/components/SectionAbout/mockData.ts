export const mockSectionAboutProps = {
  subtitle: 'infos',
  title: 'About web series',
  description:
    'Aprenda PHP do Jeito Certo Utilizando 100% de Tecnologia Moderna e os Melhores Padr√µes de Projeto Nessa Webserie 100% Online e GRATUITA!',
  topics: [
    {
      text: 'text 1 example',
      alttextIcon: 'icone',
      topicIconSrcImage: '/images/icon-eye.svg'
    },
    {
      text: 'text 2 example',
      alttextIcon: 'icone',
      topicIconSrcImage: '/images/icon-key.svg'
    },
    {
      text: 'text 3 example',
      alttextIcon: 'icone',
      topicIconSrcImage: '/images/icon-circle.svg'
    },
    {
      text: 'text 4 example',
      alttextIcon: 'icone',
      topicIconSrcImage: '/images/icon-medal.svg'
    }
  ],
  ilustrationSrcImage: '/images/about-ilustration.svg',
  ilustrationAltText: 'Illustration about'
}
