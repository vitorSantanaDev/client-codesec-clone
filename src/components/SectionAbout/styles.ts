import styled, { css } from 'styled-components'
import { Wrapper as Subtitle } from 'components/Subtitle/styles'
import { Wrapper as Heading } from 'components/Heading/styles'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: grid;
    align-items: center;
    gap: ${theme.spacings.medium};
    grid-template-columns: 5fr 7fr;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    } ;
  `}
`
export const ContentWrapperLeft = styled.div`
  ${({ theme }) => css`
    ${Subtitle} {
      margin-bottom: calc(${theme.spacings.small} - 0.8rem);
    }

    ${Heading} {
      max-width: 328px;
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`
export const ContentWrapperRight = styled.div`
  ${() => css``}
`
export const Description = styled.div`
  ${({ theme }) => css`
    line-height: 150%;
    color: ${theme.colors.gray7};
    font-size: ${theme.font.sizes.small};

    margin-bottom: ${theme.spacings.medium};
  `}
`
export const TopicsWrapper = styled.div`
  ${() => css``}
`
export const Topic = styled.span`
  ${({ theme }) => css`
    font-weight: bold;
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.white};

    display: flex;
    align-items: center;
    margin-bottom: calc(${theme.spacings.medium} - 0.4rem);
  `}
`
export const TopicIcon = styled.img`
  ${({ theme }) => css`
    width: 32px;
    height: 32px;

    margin-right: ${theme.spacings.small};
  `}
`
export const Ilustration = styled.img`
  ${() => css`
    width: 100%;
  `}
`
