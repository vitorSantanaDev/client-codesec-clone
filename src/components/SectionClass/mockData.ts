export const mockDataSectionClassProps = {
  title: 'O que você vai aprender',
  subtitle: 'Aulas',
  classs: [
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    },
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    },
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    },
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    },
    {
      dateClass: '14/03',
      hourAndDay: 'Terça as - 20:30H',
      classNumber: 'Aula 01',
      icon: {
        url: 'https://res.cloudinary.com/vsantana/image/upload/v1650500029/icon_lock_6d7fca3f3c.svg',
        altTextIcon: 'Icon'
      },
      titleClass: 'Hackeando tudo',
      descriptionClass:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Est mus nisi, sed in tempor. Nisl sit iaculis nulla consectetur. Purus egestas sit eget viverra. Nisi id in porta nec, a a. Eu in dis et, morbi ut at pellentesque. Massa eu amet morbi eu diam ultricies tellus aliquet. Vel, amet dolor cursus lorem quis amet sed. Mollis diam a purus nulla non, mattis proin odio. Facilisis molestie nibh egestas lacus adipiscing.'
    }
  ]
}
