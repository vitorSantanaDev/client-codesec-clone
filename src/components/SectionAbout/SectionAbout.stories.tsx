import { ComponentStory, ComponentMeta } from '@storybook/react'
import SectionAbout from '.'
import { mockSectionAboutProps } from './mockData'

export default {
  title: 'SectionAbout',
  component: SectionAbout,
  args: {
    ...mockSectionAboutProps
  }
} as ComponentMeta<typeof SectionAbout>

export const Default: ComponentStory<typeof SectionAbout> = (args) => (
  <SectionAbout {...args} />
)
