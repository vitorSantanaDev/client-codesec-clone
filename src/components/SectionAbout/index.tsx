import Container from 'components/Container'
import Heading from 'components/Heading'
import Subtitle from 'components/Subtitle'
import * as S from './styles'

type SectionAboutProps = {
  subtitle: string
  title: string
  description: string
  topics: {
    text: string
    alttextIcon: string
    topicIconSrcImage: string
  }[]
  ilustrationSrcImage: string
  ilustrationAltText: string
}

const SectionAbout = ({
  subtitle,
  title,
  description,
  topics,
  ilustrationSrcImage,
  ilustrationAltText
}: SectionAboutProps) => (
  <Container>
    <S.Wrapper>
      <S.ContentWrapperLeft>
        <Subtitle>{subtitle}</Subtitle>
        <Heading>{title}</Heading>
        <S.Description>{description}</S.Description>
        <S.TopicsWrapper>
          {topics.map((topic, index) => {
            const key = `${index}-${topic}`
            return (
              <S.Topic key={key}>
                {' '}
                <S.TopicIcon
                  src={topic.topicIconSrcImage}
                  alt={topic.alttextIcon}
                />
                {topic.text}
              </S.Topic>
            )
          })}
        </S.TopicsWrapper>
      </S.ContentWrapperLeft>
      <S.ContentWrapperRight>
        <S.Ilustration src={ilustrationSrcImage} alt={ilustrationAltText} />
      </S.ContentWrapperRight>
    </S.Wrapper>
  </Container>
)

export default SectionAbout
