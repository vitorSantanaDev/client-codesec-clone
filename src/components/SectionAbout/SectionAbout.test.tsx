import { screen } from '@testing-library/react'
import { renderTheme } from 'styles/theme/render-theme'
import SectionAbout from '.'
import { mockSectionAboutProps } from './mockData'

describe('<SectionAbout />', () => {
  it('should render section about with correct elements', () => {
    renderTheme(<SectionAbout {...mockSectionAboutProps} />)
    const subtitle = screen.getByText(/infos/i)
    const heading = screen.getByRole('heading', { name: /about web series/i })

    expect(subtitle).toBeInTheDocument()
    expect(heading).toBeInTheDocument()
  })
  it('should render the correct amount of topics', () => {
    renderTheme(<SectionAbout {...mockSectionAboutProps} />)
    const topics = screen.getAllByText(/text/i)
    const icons = screen.getAllByRole('img', { name: /icone/i })

    expect(topics).toHaveLength(4)
    expect(icons).toHaveLength(4)
  })
  it('should render the illustration of section about', () => {
    renderTheme(<SectionAbout {...mockSectionAboutProps} />)
    const illustration = screen.getByRole('img', {
      name: /illustration about/i
    })

    expect(illustration).toBeInTheDocument()
  })
  it('should to match snapshot', () => {
    const { container } = renderTheme(
      <SectionAbout {...mockSectionAboutProps} />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
