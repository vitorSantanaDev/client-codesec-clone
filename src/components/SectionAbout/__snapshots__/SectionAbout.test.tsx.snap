// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionAbout /> should to match snapshot 1`] = `
.c0 {
  width: 100%;
  margin: 0 auto;
  max-width: 1200px;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  font-size: 5.6rem;
  color: white;
  font-family: 'Raleway',sans-serif;
}

.c4 {
  background-image: linear-gradient(243.46deg,#FA5770 16.66%,#E83752 83.34%);
  background-clip: text;
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 700;
  -webkit-letter-spacing: 15%;
  -moz-letter-spacing: 15%;
  -ms-letter-spacing: 15%;
  letter-spacing: 15%;
  text-transform: uppercase;
  font-size: calc(1.6rem - 0.2rem);
}

.c1 {
  height: 90vh;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2.4rem;
  grid-template-columns: 5fr 7fr;
}

.c2 .c3 {
  margin-bottom: calc(1.6rem - 0.8rem);
}

.c2 .c5 {
  max-width: 328px;
  margin-bottom: 2.4rem;
}

.c7 {
  line-height: 150%;
  color: #A3A3A3;
  font-size: 1.6rem;
  margin-bottom: 2.4rem;
}

.c8 {
  font-weight: bold;
  font-size: 1.6rem;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: calc(2.4rem - 0.4rem);
}

.c9 {
  width: 32px;
  height: 32px;
  margin-right: 1.6rem;
}

.c10 {
  width: 100%;
}

@media (max-width:768px) {
  .c6 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c1 {
    grid-template-columns: 1fr;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <span
        class="c3 c4"
        color="redGradient"
      >
        infos
      </span>
      <h1
        class="c5 c6"
        color="white"
      >
        About web series
      </h1>
      <div
        class="c7"
      >
        Aprenda PHP do Jeito Certo Utilizando 100% de Tecnologia Moderna e os Melhores Padr√µes de Projeto Nessa Webserie 100% Online e GRATUITA!
      </div>
      <div
        class=""
      >
        <span
          class="c8"
        >
           
          <img
            alt="icone"
            class="c9"
            src="/images/icon-eye.svg"
          />
          text 1 example
        </span>
        <span
          class="c8"
        >
           
          <img
            alt="icone"
            class="c9"
            src="/images/icon-key.svg"
          />
          text 2 example
        </span>
        <span
          class="c8"
        >
           
          <img
            alt="icone"
            class="c9"
            src="/images/icon-circle.svg"
          />
          text 3 example
        </span>
        <span
          class="c8"
        >
           
          <img
            alt="icone"
            class="c9"
            src="/images/icon-medal.svg"
          />
          text 4 example
        </span>
      </div>
    </div>
    <div
      class=""
    >
      <img
        alt="Illustration about"
        class="c10"
        src="/images/about-ilustration.svg"
      />
    </div>
  </div>
</div>
`;
